### ISSUE TITLE
Substitution does not work after ParamSpec substitution of the user generic with a list of TypeVars

### ISSUE THREAD

**Issue by serhiy-storchaka** [python/cpython#88965]
BPO | [44802](https://bugs.python.org/issue44802)
--- | :---
Nosy | @gvanrossum, @serhiy-storchaka, @Fidget-Spinner

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = None
created_at = <Date 2021-08-01.11:37:51.360>
labels = ['type-bug', 'library', '3.10', '3.11']
title = 'Substitution does not work after ParamSpec substitution of the user generic with a list of TypeVars'
updated_at = <Date 2021-08-01.11:37:51.360>
user = 'https://github.com/serhiy-storchaka'
```

bugs.python.org fields:
```python
activity = <Date 2021-08-01.11:37:51.360>
actor = 'serhiy.storchaka'
assignee = 'none'
closed = False
closed_date = None
closer = None
components = ['Library (Lib)']
creation = <Date 2021-08-01.11:37:51.360>
creator = 'serhiy.storchaka'
dependencies = []
files = []
hgrepos = []
issue_num = 44802
keywords = []
message_count = 1.0
messages = ['398694']
nosy_count = 3.0
nosy_names = ['gvanrossum', 'serhiy.storchaka', 'kj']
pr_nums = []
priority = 'normal'
resolution = None
stage = None
status = 'open'
superseder = None
type = 'behavior'
url = 'https://bugs.python.org/issue44802'
versions = ['Python 3.10', 'Python 3.11']
```

</p></details>

### Linked PRs
* gh-102808
* gh-102957

**Comment by LeeeeT** [comment]
Hi there! I'm experiencing the same issue. Will this be fixed? @serhiy-storchaka

**Comment by AlexWaygood** [comment]
Fixed on `main` and 3.11, but we've decided not to backport to 3.10.

**Comment by LeeeeT** [comment]
Thanks! You guys are awesome! ❤️
